artifacttemplates/http%3A%2F%2Fopentosca.org%2Fartifacttemplates/Ubuntu-VM_OperatingSystemInterface-w1/source/src/main/java/org/opentosca/artifacttemplates/ubuntuvm/VirtualMachine.java

```diff
@@ -0,0 +1,204 @@
+package org.opentosca.artifacttemplates.ubuntuvm;
+
+import java.io.File;
+import java.io.InputStream;
+
+import com.jcraft.jsch.ChannelExec;
+import com.jcraft.jsch.ChannelSftp;
+import com.jcraft.jsch.JSch;
+import com.jcraft.jsch.Session;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class VirtualMachine {
+    private static final Logger LOG = LoggerFactory.getLogger(VirtualMachine.class);
+
+    private final String host;
+    private final int port = 22;
+    private final String user;
+    private final String key;
+    private Session session = null;
+
+    public VirtualMachine(String host, String user, String key) {
+        this.host = host;
+        this.user = user;
+        this.key = key;
+    }
+
+    public void connect() throws Exception {
+        LOG.info("Creating session on VM '{}'", this);
+
+        if (host.equals("TESTMODE")) {
```

Comment:
We should avoid such things... This can be easily mocked using a mocking server, as done, e.g., for [OpenStack](https://github.com/OpenTOSCA/tosca-definitions-common/blob/main/artifacttemplates/http%253A%252F%252Fopentosca.org%252Fartifacttemplates/OpenStack_CloudProviderInterfaceIA-w2/source/src/test/java/org/opentosca/artifacttemplates/openstack/OpenStackTest.java)

Reply:
The pressure of two thumbs down is to heavy 😭 

I tried to write some test that stubs VirtualMachine::connect and SoapUtils.
See [WaitForAwailabilityTest](https://github.com/OpenTOSCA/tosca-definitions-common/blob/cf5968ce0f45d48acc709029fc6e752e1c22c1ff/artifacttemplates/http%253A%252F%252Fopentosca.org%252Fartifacttemplates/Ubuntu-VM_OperatingSystemInterface-w1/source/src/test/java/org/opentosca/artifacttemplates/ubuntuvm/WaitForAvailabilityTest.java).

What do you think? 

Getting this to work was an odyssey 🌊
E.g. I needed to add these lines to the pom.

```
<plugin>
    <artifactId>maven-surefire-plugin</artifactId>
    <configuration>
        <argLine>--add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED</argLine>
    </configuration>
</plugin>
```

Btw, the test is not executed in the pipeline.
I think thats because of `org.junit.Test` instead of `org.junit.jupiter.api.Test`.

Reply:
Yes please use JUnit 5 instead of 4.

Reply:
I actually do not know how. I dont see any junit5 support for powermock?

Reply:
I still wonder why you are using powermock for this. You don't test the code of the VM class. I thought about something like [MockSshServer](https://github.com/eclipse/winery/blob/ef5daece4a9e9184b7a37962a49f5a21d85314da/org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/TestWithGitRepoAndSshServer.java#L35).

Reply:
done